### PR TITLE
Change non-homepage HTML titles to be "Page-title &ndash; gRPC"

### PR DESCRIPTION
**Pull request by chalin**
- This follows a common convention used, for example, on https://kubernetes.io.
- Prep work for #363, and contribution to #362.

Preview: https://deploy-preview-364--grpc-io.netlify.app/

cc @ejona86

### CODE REVIEW

File: layouts/blog/single.html [hunk 1]
```diff
@@ -1,8 +1,8 @@
-{{ define "title" }}
-The gRPC Blog &ndash; {{ .Title }}
-{{ end }}
+{{ define "title" -}}
+{{ .Title }} &ndash; Blog | {{ site.Title -}}
```
Comment (chalin) [inline layouts/blog/single.html:2]: Note the little variance in the title for blog entries: "page-title - Blog | gRPC"

Comment (chalin) [reply layouts/blog/single.html]: I've removed "Blog"

**Comment by chalin** [comment]
Following @ejona86's feedback, we're now using `&ndash;` consistently as the separator.

**Comment by chalin** [comment]
@ejona86 - thanks for the feedback

**Comment by ejona86** [comment]
Some reasoning: I pointed out that Gmail and Github both use the same separator multiple times. I wanted to keep " - " simply because that's what we were using already (to avoid bikeshedding and flip/flopping every year). Chrome's window itself uses a " - " on my machine (not for tabs, but the window), which is just another instance of using the same separator multiple times. In all cases though it seems the site name is _last_ in the URL, not first like we had previously; the specific page is first so that it is visible (instead of all your tabs saying "gRPC - Q..."

Aside: I noticed that '-' seems to be much more common at the moment than '|', at least in the sites I had open at the time (but that basically means "various Google properties" "Github" "Wikipedia"... I learned something about myself).

**Pull request merged**